### PR TITLE
fix Pointer is missing a nullability type specifier (_Nonnull, _Nulla…

### PR DESCRIPTION
**Pull request by wfanforever**
…ble, or _Null_unspecified)

### CODE REVIEW

**Comment by jamesward** [comment]
@wfanforever I've sent you an invite to the org: https://github.com/forcedotcom

Once accepted, go refresh the status of the CLA bot at: https://cla.salesforce.com/status/forcedotcom/SalesforceMobileSDK-iOS/pull/2546

File: libs/SmartSync/SmartSync/Classes/Model/SFLayout.h [hunk 1]
```diff
@@ -77,7 +77,7 @@
  * @param data JSON data.
  * @return Instance of this class.
  */
-+ (instancetype)fromJSON:(NSDictionary *)data;
++ (instancetype)fromJSON:(nonnull NSDictionary *)data;
```
Comment (khawkins) [inline libs/SmartSync/SmartSync/Classes/Model/SFLayout.h]: These return types need nullability markers too, yes?

Comment (wfanforever) [reply libs/SmartSync/SmartSync/Classes/Model/SFLayout.h]: ugh yes, let me update all them..

**Comment by wfanforever** [comment]
@jamesward any idea why the CLA check still failed? 🤔

File: libs/SmartSync/SmartSync/Classes/Manager/SFLayoutSyncManager.h [hunk 2]
```diff
@@ -38,7 +38,7 @@
  * @param objectType Object type.
  * @param layout Layout.
  */
-typedef void (^SFLayoutSyncCompletionBlock) (NSString *objectType, SFLayout *layout);
+typedef void (^SFLayoutSyncCompletionBlock) (NSString * _Nonnull objectType, SFLayout  * _Nonnull layout);
```
Comment (bhariharan) [inline libs/SmartSync/SmartSync/Classes/Manager/SFLayoutSyncManager.h]: The specifier for `SFLayout *` should be `_Nullable`.

File: libs/SmartSync/SmartSync/Classes/Manager/SFMetadataSyncManager.h [hunk 3]
```diff
@@ -37,7 +37,7 @@
  *
  * @param metadata Metadata.
  */
-typedef void (^SFMetadataSyncCompletionBlock) (SFMetadata *metadata);
+typedef void (^SFMetadataSyncCompletionBlock) (SFMetadata  * _Nonnull metadata);
```
Comment (bhariharan) [inline libs/SmartSync/SmartSync/Classes/Manager/SFMetadataSyncManager.h]: The specifier for `SFMetadata *` should be `_Nullable`.

**Comment by wfanforever** [comment]
Thanks @bhariharan ! updated.

**Comment by jamesward** [comment]
@wfanforever The CLA bot is operating normally again.

**Pull request merged**